examples/advanced/Tutorial3/reconstruction/FairTestDetectorRecoTask.cxx

```diff
@@ -97,7 +97,9 @@ void FairTestDetectorRecoTask::Exec(Option_t* /*opt*/)
         TVector3 dpos(1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12), 1 / TMath::Sqrt(12));
 
         FairTestDetectorHit* hit = new ((*fHitArray)[ipnt]) FairTestDetectorHit(-1, -1, pos, dpos);
-        hit->AddLink(FairLink(-1, FairRootManager::Instance()->GetEntryNr(), FairRootManager::Instance()->GetBranchId("FairTestDetectorDigi"), ipnt));
+        if ( FairRootManager::Instance() != NULL ) {
```

Comment:
@MohammadAlTurany The caller of `Instance()` should really not worry about the existance of the singleton object. Can we rather fix the singleton implementation of `FairRootManager`? Best would be to return a reference, but that will break existing code. The default constructor of `FairRootManager` should also be private, but that may also break existing code, I assume.

The best non-breaking solution I can think of is
```cpp
FairRootManager* FairRootManager::Instance()
{
// Returns singleton instance.
  if (fgInstance == nullptr)
  {
    fgInstance = new FairRootManager();
  }
  return fgInstance;
}
```

But even with the above change the code is not safe. The initialization in the ctor `fgInstance = this;` is dangerous. For example, if the caller of the ctor puts the object on a stack which goes out of scope before the program exits, each call to `FairRootManager::Instance()` will eventually point to random memory.